app/helpers/helpers.rb

```diff
@@ -0,0 +1,11 @@
+class Helpers
+  
+
+    def self.current_user(session)
+      @user = User.find(session[:user_session_id])
```

Comment:
Ah, that makes so much sense.  

I made the is_logged_in? and current_user methods for a lab in Sinatra waaaaaay before and the way I got it to work just stayed the way I worked it.  

So now my helper methods in my sinatra project look all goofy too.  But this is way cleaner (and I see the later comments about access to session).

Thanks for this!

> On Apr 9, 2016, at 7:41 PM, Randall Reed <notifications@github.com> wrote:
> 
> In app/helpers/helpers.rb <https://github.com/learn-co-students/sinatra-fwitter-group-project-v-000/pull/79#discussion_r59125699>:
> 
> > @@ -0,0 +1,11 @@
> > +class Helpers
> > +  
> > +
> > +    def self.current_user(session)
> > +      @user = User.find(session[:user_session_id])
> Because in the lifespan of a request, you might call current_user multiple times, we'd prefer this method to only hit the database once and then 'cache' the user instance into an instance variable that could be returned on any subsequent call to current_user.
> 
> def current_user
>   @current_user ||= User.find(session[:id) if session[:id]
> end
> This pattern of conditionally setting an instance variable or returning it if it's already defined using the ||= (or-equals) operator is called Memoization (read more <http://gavinmiller.io/2013/basics-of-ruby-memoization/> and here <http://www.justinweiss.com/articles/4-simple-memoization-patterns-in-ruby-and-one-gem/>).
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/learn-co-students/sinatra-fwitter-group-project-v-000/pull/79/files/9dfc20de18dce246a6925028c350c9316e9e3937#r59125699>